modules/API/external/external_api.js

```diff
@@ -84,323 +70,342 @@ function changeParticipantNumber(APIInstance, number) {
         = configToURLParamsArray(interfaceConfigOverwrite);
+
     if (interfaceConfigURLParams.length) {
-        this.url += '&interfaceConfig.'
-            + interfaceConfigURLParams.join('&interfaceConfig.');
+        url += `&interfaceConfig.${
+            interfaceConfigURLParams.join('&interfaceConfig.')}`;
     }
 
-    this.frame = document.createElement("iframe");
-    this.frame.src = this.url;
-    this.frame.name = this.frameName;
-    this.frame.id = this.frameName;
-    this.frame.width = "100%";
-    this.frame.height = "100%";
-    this.frame.setAttribute("allowFullScreen","true");
-    this.frame = this.iframeHolder.appendChild(this.frame);
-    this.postis = postisInit({
-        window: this.frame.contentWindow,
-        scope: "jitsi_meet_external_api_" + id
-    });
-
-    this.eventHandlers = {};
-
-    // Map<{string} event_name, {boolean} postis_listener_added>
-    this.postisListeners = {};
-
-    this.numberOfParticipants = 1;
-    this._setupListeners();
-
-    id++;
+    return url;
 }
 
 /**
- * Executes command. The available commands are:
- * displayName - sets the display name of the local participant to the value
- * passed in the arguments array.
- * toggleAudio - mutes / unmutes audio with no arguments
- * toggleVideo - mutes / unmutes video with no arguments
- * filmStrip - hides / shows the film strip with no arguments
- * If the command doesn't require any arguments the parameter should be set
- * to empty array or it may be omitted.
- * @param name the name of the command
- * @param arguments array of arguments
+ * The IFrame API interface class.
  */
-JitsiMeetExternalAPI.prototype.executeCommand
-= function(name, ...argumentsList) {
-    if(!(name in commands)) {
-        logger.error("Not supported command name.");
-        return;
+class JitsiMeetExternalAPI extends EventEmitter {
+    /**
+     * Constructs new API instance. Creates iframe and loads Jitsi Meet in it.
+     *
+     * @param {string} domain - The domain name of the server that hosts the
+     * conference.
+     * @param {string} [roomName] - The name of the room to join.
+     * @param {number} [width] - Width of the iframe.
+     * @param {number} [height] - Height of the iframe.
+     * @param {DOMElement} [parentNode] - The node that will contain the
+     * iframe.
+     * @param {Object} [configOverwrite] - Object containing configuration
+     * options defined in config.js to be overridden.
+     * @param {Object} [interfaceConfigOverwrite] - Object containing
+     * configuration options defined in interface_config.js to be overridden.
+     * @param {boolean} [noSSL] - If the value is true https won't be used.
+     * @param {string} [jwt] - The JWT token if needed by jitsi-meet for
+     * authentication.
+     */
+    constructor(domain, // eslint-disable-line max-params
+        roomName = '', width = MIN_WIDTH, height = MIN_HEIGHT,
+        parentNode = document.body, configOverwrite = {},
```

Comment:
Nope, I don't have other ideas or issues I can think of. I just noticed most of this PR tried to avoid modifying existing behavior but this changed it a little.